src/quo/components/buttons/slide_button/view.cljs

```diff
@@ -54,9 +54,10 @@
         on-track-layout               (rn/use-callback
                                        #(set-track-width (oops/oget % "nativeEvent.layout.width")))
         reset-fn                      (rn/use-callback
-                                       (fn []
+                                       (fn [reset-to-end-position?]
                                          (set-sliding-complete false)
-                                         (animations/reset-track-position x-pos)))
+                                         (animations/reset-track-position
+                                          (if reset-to-end-position? track-width x-pos))))
```

Comment:
Do we always want to reset the x-position when we reset the slider? For example, what should happen if we fail to authenticate and we need to reset the slider?

Reply:
From what I see, it would reset it to the "end" only when successful, and if failed, it would go back to the initial `x-pos`. Correct me if I'm wrong @alwx.

Reply:
The idea here is that the tracker has two positions — the zero position and the end position, and everything else is basically a transformation from one position to another.
In the case of a failure we want to move back to the zero position, and in the case of a success we want to move to the end position (we should already be there but we do it for extra safety). That's basically what this code does. If `reset-to-end-position?` is true then we move it to the end position; we move it to the zero position otherwise.

Reply:
I suppose what I'm thinking is that we could only call the reset function when we know we want to reset from the end-position to the start-position. But I'm not sure how many changes that would require to make that possible, thoughts?